library/src/c_api/mod.rs

```diff
@@ -183,6 +194,20 @@ fn path_to_c_string(path: Option<PathBuf>) -> anyhow::Result<*mut c_char> {
     })
 }
 
+fn to_update_result(status: anyhow::Result<UpdateStatus>) -> UpdateResult {
+    let result = match status {
+        Ok(status) => UpdateResult {
+            status: status as i32,
+            message: std::ptr::null(),
+        },
```

Comment:
Do you want this to return std::ptr::null() (for the whole object) when there is no update?

Reply:
No because we have a status that represents no update as the failure reason.